metaflow/plugins/events_decorator.py

```diff
@@ -188,6 +241,63 @@ def flow_init(
 
         # TODO: Handle scenario for local testing using --trigger.
 
+    def format_deploytime_value(self):
+        for trigger in self.triggers:
+            # Case were trigger is a function that returns a list of events
+            # Need to do this bc we need to iterate over list later
+            if isinstance(trigger, DeployTimeField):
+                if isinstance(deploy_time_eval(trigger), list):
+                    deploy_val = deploy_time_eval(trigger)
+                    self.triggers.remove(trigger)
+                    self.triggers.extend(deploy_val)
+            else:
+                break
+        for trigger in self.triggers:
+            old_trigger = trigger
+            # Entire event is a function (returns either string or dict)
+            if isinstance(trigger, DeployTimeField):
+                trigger = deploy_time_eval(trigger)
+                try:
+                    trigger = json.loads(trigger)
+                except (TypeError, json.JSONDecodeError):
+                    pass
+                # Case where just the name is a function (always a str)
+                if isinstance(trigger, str):
+                    trigger = {"name": trigger}
+            trigger_params = trigger.get("parameters", {})
+            # Case where param is a function (can return list or dict)
+            if isinstance(trigger_params, DeployTimeField):
+                trigger_params = deploy_time_eval(trigger_params)
+                try:
+                    trigger_params = json.loads(trigger_params)
```

Comment:
we shouldn't need this -- the user needs to return this appropriately.

Reply:
I think this happens when the user returns a dictionary, not sure why